### ISSUE TITLE
Feature Request: Adventure modules

### ISSUE THREAD

**Issue by l9f** [astranauta/astranauta.github.io#493]
Have you guys considered adding the adventure modules, with maps, all nice and hyperlinked?

**Comment by TheGiddyLimit** [comment]
That's awfully vague - could you please explain exactly what you're imagining?

**Comment by l9f** [comment]
Like adding an "Adevnture"tab on the ui, with "Lost Mines" as a drop down, and then have the whole adventure text, hyperlinked and with maps etc...

**Comment by TheGiddyLimit** [comment]
Hm, fair enough. Certainly possible, but it would take a mind-boggling amount of work, for something I feel isn't much better than a bookmarked PDF? To implement this properly would _probably_ require a complete re-write of a great deal of the site, maybe. Convince me on the merits of the idea, though, and I might look into it?

**Comment by l9f** [comment]
Well, then maybe it's a separate html file like `/adventure.html`  instead of a tab on `5etools.html` to avoid a complete rewrite. I'm just thinking that RollTwenty and Beyond have web page versions of the adventures that seem much better than bookmarked PDFs. And you already have the backend with the monsters and items etc..  

https://www.youtube.com/watch?v=APm1IzaHOjA as an example; (Not the map / VTT features itself, but see how they do the adventure text that is all hyperlinked and jumps around?)

**Comment by TheGiddyLimit** [comment]
I more meant a complete rewrite in terms of linking things together nicely, but what I was envisioning is complete overkill. 

I have a couple adventures on Roll20; if we went with that style this could actually be pretty easy, assuming we could get a hold of the necessary data (i.e. someone who's purchased this stuff on either Roll20 or Beyond helps us rip it).

Roll20 is a funny example though, because it's genuinely horrible to use in terms of navigating/reading the adventure (imo). I imagine this wouldn't be a problem if we formatted things nicely.

So, essentially, give me a data dump of all the adventures, and your dream can be reality. Maybe.

**Comment by l9f** [comment]
Beyond would probably be the better example since it doesn't have to fit Roll20's arbitrary 'document' item in the sidebar thingy.

**Comment by l9f** [comment]
So it's not as simple as defining links like `{@creature Goblin/MM}` or some syntax like that for monsters and spells? For internal links in the document itself, you could either do htmlcontent in the data and hand roll `<p name="linktarget"> FOO</p>` stuff.  

You know what make a good thing for that is to get it into a sphinx document or something like one of the auto doc generators. 

I was just thinking out loud. :)

**Comment by TheGiddyLimit** [comment]
Yeah, as I said, should actually be pretty easy to do, provided I (we) have the data. Which I certainly don't, and don't want to pay $$$ to unlock it all on Beyond. I'm sure somebody owns this junk, but I'd have to track them down, and spend a great deal of time, frankly, pissing around to get my grubby hands on what I needed. If you own an adventure on Beyond and want to help figure this out and perhaps use it to make a prototype page, I'm open to the idea.

**Comment by l9f** [comment]
I don't have a Beyond adventure, but I can upload an example data file of LMoP or something to prototype / decide on data format and then we can get community to do the builk manual work?

**Comment by TheGiddyLimit** [comment]
Meh, I'll just pay the $10 for LMoP. I'll need it anyway, so I can write a script to export an adventure, and then find people who own the others. Ought to be the easiest way of doing it. I've a few other pieces of work I need to get through first, though, so I'll get back to you - hopefully sometime this coming week.

**Comment by l9f** [comment]
Fair enough. I haven't tried buying anything on Beyond yet.
